WordPress/gutenberg
# [RNMobile] Use RichText component on Title block

This PR replace the `TextInput` component used as base layer for the Title block in favor of `RichText` that does have better handling of the Enter.key pressed event.

On a previous PR we've made the Title block with multiline=false prop, and detected the Enter.key with the default `onSubmitEditing` prop, given by the underlying TextInput RN component.

Despite `onSubmitEditing` is a very reliable way of detecting Enter.key on TextInput with multiline = false, and works on both HW and SW keyboard without the necessity of implementing hacks with TextWatchers and keyListeners - https://facebook.github.io/react-native/docs/textinput#onsubmitediting it does have some problems on Android where the Keyboard is dismissed, and there is no a simple way to get it back without seeing it down and immediately up for the end-user.

(We've had the same problem on `Code` and `More` block, and disabled them for the alpha, since the UX was not ideal).

To test:

- Start the mobile demo app
- Put the caret somewhere in the Title block
- Tap enter on the soft keyboard, or on the hw keyboard
- It should add a new empty (para) block after the current


Mobile side PR is here: https://github.com/wordpress-mobile/gutenberg-mobile/pull/513
